lib/sidekiq.rb

```diff
@@ -149,7 +149,12 @@ def self.default_server_middleware
   end
 
   def self.default_worker_options=(hash)
-    @default_worker_options = default_worker_options.merge(hash.stringify_keys)
+    h = hash.dup
+    # stringify
+    h.keys.each do |key|
+      h[key.to_s] = h.delete(key)
+    end
+    @default_worker_options = default_worker_options.merge(h)
```

Comment:
If you're still worried about performance, I tried both approaches in IRB with 100 symbol keys in the array. The both seem to take something like a bit less than 10us... Just give it a try if you don't believe ;)